extensions/ql-vscode/test/vscode-tests/no-workspace/databases/local-databases-ui.test.ts

```diff
@@ -7,14 +7,57 @@ import {
   createFileSync,
   pathExistsSync,
 } from "fs-extra";
-import { Uri } from "vscode";
+import { Uri, window } from "vscode";
+import type { DatabaseSelectionQuickPickItem } from "../../../../src/databases/local-databases-ui";
 
 import { DatabaseUI } from "../../../../src/databases/local-databases-ui";
 import { testDisposeHandler } from "../../test-dispose-handler";
 import { createMockApp } from "../../../__mocks__/appMock";
 import { QueryLanguage } from "../../../../src/common/query-language";
+import { mockedQuickPickItem, mockedObject } from "../../utils/mocking.helpers";
 
 describe("local-databases-ui", () => {
+  const storageDir = dirSync({ unsafeCleanup: true }).name;
+  const db1 = createDatabase(storageDir, "db1-imported", QueryLanguage.Cpp);
+  const db2 = createDatabase(storageDir, "db2-notimported", QueryLanguage.Cpp);
+  const db3 = createDatabase(storageDir, "db3-invalidlanguage", "hucairz");
+
+  // these two should be deleted
+  const db4 = createDatabase(
+    storageDir,
+    "db2-notimported-with-db-info",
+    QueryLanguage.Cpp,
+    ".dbinfo",
+  );
+  const db5 = createDatabase(
+    storageDir,
+    "db2-notimported-with-codeql-database.yml",
+    QueryLanguage.Cpp,
+    "codeql-database.yml",
+  );
+
+  const app = createMockApp({});
+  const databaseUI = new DatabaseUI(
+    app,
+    {
+      databaseItems: [{ databaseUri: Uri.file(db1) }],
+      onDidChangeDatabaseItem: () => {
+        /**/
+      },
+      onDidChangeCurrentDatabaseItem: () => {
+        /**/
+      },
+      setCurrentDatabaseItem: () => {},
+    } as any,
+    {
+      onLanguageContextChanged: () => {
+        /**/
+      },
+    } as any,
+    {} as any,
+    storageDir,
+    storageDir,
+  );
```

Comment:
I moved this code up from the "orphaned databases" test, since I needed access to the same `databaseUI` and example data and didn't want to duplicate the code. I don't think this is a problem, but happy to change approach otherwise!